### PR TITLE
Make latency spikes less likely to happen

### PR DESCRIPTION
**Pull request by mocchira**
At the previous attempt, PR didn't get merged because Segfault had happened on some platform.
This PR might solve the Segfault by avoiding unnecessary GC invocations so please give it one more try. > @yosukehara @windkit

### CODE REVIEW

**Comment by windkit** [comment]
The unit test is failing for OTP 18.3 with segfault.

To me, it indicates some native code is having problems with memory de/alloc

**Comment by mocchira** [comment]
@windkit what exact version did you use? if 18.3.4.5 then it had some serious issues that can cause segfault so try with 18.3.4.7. if you tried 18.3.4.7 then I'll close.

**Comment by windkit** [comment]
@mocchira so may be we have to update the `.travis.yml` for that?

File: src/leo_object_storage_server.erl [hunk 1]
```diff
@@ -430,9 +430,12 @@ handle_call({put = Method, #?OBJECT{addr_id = AddrId,
     Key_1 = ?gen_backend_key(StorageInfo#backend_info.avs_ver_cur,
                              AddrId, Key),
     {Reply, State_1} = put_1(Key_1, Object, State),
-
-    erlang:garbage_collect(self()),
-    reply(Method, Key, Reply, InTime, State_1);
+    case State#obj_server_state.is_gc_wip of
+        false ->
+            erlang:garbage_collect(self(), [{async, void}]);
```
Comment (yosukehara) [inline src/leo_object_storage_server.erl]: I've recognized we cannot use `erlang:garbage_collect/2` due to https://github.com/leo-project/leo_object_storage/pull/17#issuecomment-348422811

Comment (mocchira) [reply src/leo_object_storage_server.erl]: I guess multiple invocations of erlang:garbage_collect/2 might hit the VM crash problem so this PR can avoid the problem by making it happen on only one process.

**Comment by mocchira** [comment]
@windkit Unfortunately travis CI doesn't allow us to specify fine grained otp versions more than what kerl provides as described here: https://docs.travis-ci.com/user/languages/erlang/

**Comment by yosukehara** [comment]
@mocchira I've tested your PR with Erlang/OTP 18.3.4.7 and faced the same situation with #17 as below:

```
$ make
.
.
.
[os_mon] cpu supervisor port (cpu_sup): Erlang has closed
[os_mon] memory supervisor port (memsup): Erlang has closed
Makefile:20: recipe for target 'eunit' failed
make: *** [eunit] Segmentation fault (core dumped)

$ erl
Erlang/OTP 18 [erts-7.3.1.4] [source] [64-bit halfword] [smp:4:4] [async-threads:10] [kernel-poll:false]

```

**Comment by mocchira** [comment]
@yosukehara Thanks for confirming. I'll close the PR. In future, it might be good to revisit this PR once otp18 get deprecated.

**Comment by yosukehara** [comment]
@mocchira 

> it might be good to revisit this PR once otp18 get deprecated.

Agree

**Pull request closed without merging**